dspace-jspui/src/main/java/org/dspace/app/webui/components/VersioningItemHome.java

```diff
@@ -91,26 +99,49 @@ public void process(Context context, HttpServletRequest request,
 				throw new PluginException(e.getMessage());
 			}
 
-			if (latestVersion != null && latestVersion.getItem() != null
-					&& !latestVersion.getItem().getID().equals(item.getID())) {
-				// We have a newer version
-				Item latestVersionItem = latestVersion.getItem();
-				if (latestVersionItem.isArchived()) {
-					// Available, add a link for the user alerting him that
-					// a new version is available
-					newVersionAvailable = true;
-					try {
-						latestVersionURL = handleService.resolveToURL(
-								context, latestVersionItem.getHandle());
-					} catch (SQLException e) {
-						throw new PluginException(e.getMessage());
-					}
-					latestVersionHandle = latestVersionItem.getHandle();
-				} else {
-					// We might be dealing with a workflow/workspace item
-					showVersionWorkflowAvailable = true;
-				}
-			}
+
+                        if (latestVersion != null && latestVersion.getItem() != null
+                                        && !latestVersion.getItem().getID().equals(item.getID())) {
+                                // We have a newer version
+                                Item latestVersionItem = latestVersion.getItem();
+                                if (latestVersionItem.isArchived()) {
+                                        // Available, add a link for the user alerting him that
+                                        // a new version is available
+                                        newVersionAvailable = true;
+
+                                        // look up the the latest version handle
+                                        String latestVersionHandle = latestVersionItem.getHandle();
+
+                                        // lookup the latest version doi
+                                        String latestVersionDOI = null;
+                                        try {
+                                            latestVersionDOI = 
+                                                    doiService.DOIToExternalForm(
+                                                            identifierService.lookup(
+                                                                    context, latestVersionItem, DOI.class));
+                                        }
+                                        catch (IdentifierException ex)
+                                        {
+                                            log.info("Unable to convert DOI '" + latestVersionDOI 
+                                                    + "' into external form: " + ex.toString(), ex);
+                                            throw new PluginException(ex);
```

Comment:
Would throwing an exception here not a bit harsh, perhaps a fallback to handle is more gracious ?